### ISSUE TITLE
If/Else blocks throwing

### ISSUE THREAD

**Issue by Zyst** [jemc/js-abstrate#3]
It seems that `if` and `else` blocks can throw sometimes. To figure out why I first had to find a replication case, written in the test below:

```js
  it("Should render an else if block output as expected", () => {
    const html = `
{{ $Variable := "button" }}

{{ if eq $Variable "text" }}
<span>Some text goes here</span>
{{ else if eq $Variable "button" }}
<button>Button</button>
{{ else }}
<h1>Header!</h1>
{{ end }}`;

    const result = Abstrate.go.render(
      html,
      {},
      {
        variables: {},
      }
    )

    assert.equal(result, "\n\n\n\n<button>Button</button>\n")
  })
```

This will throw with an error:

```
  1) Abstrate.go.render
       Throws an error on multiple if checks:
     TypeError: nodes.map is not a function
      at Object.exports.Interpret.nodes (src/go.interpret.ts:41:16)
      at Object.exports.Interpret.body (src/go.interpret.ts:32:15)
      at Object.exports.Interpret.scope (src/go.interpret.ts:24:28)
      at Object.if (src/go.interpret.ts:122:26)
      at /home/zyst/dev/abstrate/src/go.ast.ts:180:36
      at Object.exports.Interpret.node (src/go.interpret.ts:226:19)
      at /home/zyst/dev/abstrate/src/go.interpret.ts:41:49
      at Array.map (<anonymous>)
      at Object.exports.Interpret.nodes (src/go.interpret.ts:41:16)
      at Object.exports.Interpret.body (src/go.interpret.ts:32:15)
      at Object.render (src/go.render.ts:18:22)
      at Context.<anonymous> (test/go.render.test.ts:371:32)
```

Here's some additional exploration. Marking down which elements throw, and which ones pass:

```js
  // Passes
  it("Should render the first entry of an if correctly", () => {
    const html = `
{{ if eq $Variable "text" }}
<span>Some text goes here</span>
{{ else if eq $Variable "button" }}
<button>Button</button>
{{ else }}
<h1>Header!</h1>
{{ end }}`;

    const result = Abstrate.go.render(
      html,
      {},
      {
        variables: {
          Variable: "text"
        }
      }
    );

    assert.equal(result, "\n\n<span>Some text goes here</span>\n");
  });

  // Throws
  it("Should render an else if block output as expected", () => {
    const html = `
{{ if eq $Variable "text" }}
<span>Some text goes here</span>
{{ else if eq $Variable "button" }}
<button>Button</button>
{{ else }}
<h1>Header!</h1>
{{ end }}`;

    const result = Abstrate.go.render(
      html,
      {},
      {
        variables: {
          Variable: "button"
        }
      }
    );

    assert.equal(result, "\n\n\n\n<button>Button</button>\n");
  });

  // Throws
  it("Should render an else block output as expected", () => {
    const html = `
{{ if eq $Variable "text" }}
<span>Some text goes here</span>
{{ else if eq $Variable "button" }}
<button>Button</button>
{{ else }}
<h1>Header!</h1>
{{ end }}`;

    const result = Abstrate.go.render(
      html,
      {},
      {
        variables: {
          Variable: "anything-else"
        }
      }
    );

    assert.equal(result, "\n\n<h1>Header</h1>\n");
  });
```

The reason the throw is happening is that in [Line 41 of `Interpret.nodes`](https://github.com/jemc/js-abstrate/blob/master/src/go.interpret.ts#L41) we call `nodes.map`, but the item being passed to the function is an object, which doesn't have a `map` method.

For example, the output of the following small edit:

```js
// To interpret a list of nodes, interpret each node and return the list.
Interpret.nodes = (nodes: Array<AST.Any>, data: any, runtime: IGoInterpretRuntime) => {
  // If the item isn't an array, we wish to console.log it for debugging purposes
  if (!Array.isArray(nodes)) console.log(JSON.stringify(nodes, null, 2))

  return nodes.map((node) => { return Interpret.node(node, data, runtime) })
}
```

Whilst running the tests above is:

```json
{
  "beginOffset": 75,
  "finalOffset": 166,
  "term": {
    "beginOffset": 78,
    "finalOffset": 104,
    "target": {
      "beginOffset": 78,
      "finalOffset": 80,
      "name": "eq",
      "type": "builtin"
    },
    "args": [
      {
        "beginOffset": 81,
        "finalOffset": 94,
        "name": "VariableType",
        "type": "variable"
      },
      {
        "beginOffset": 95,
        "finalOffset": 103,
        "content": "button",
        "type": "string"
      }
    ],
    "type": "invoke"
  },
  "body": [
    {
      "beginOffset": 106,
      "finalOffset": 131,
      "content": "\n<button>Button</button>\n",
      "type": "text"
    }
  ],
  "elseBody": [
    {
      "beginOffset": 141,
      "finalOffset": 159,
      "content": "\n<h1>Header!</h1>\n",
      "type": "text"
    }
  ],
  "type": "if"
}
```

**Comment by jemc** [comment]
Thanks for the detailed issue ticket - it made fixing this quite trivial - it took only ~10m to make the test and fix.

Should be fixed by 1d7b5068ce3b060fd1128709fa1a0e7a5491e77d

Basically, what I did was fix the parser to wrap the `else if` body in an array, so that it is consistent with what is expected in the interpreter for all `else` bodies.

Please let me know if you hit more issues.

**Comment by himedlooff** [comment]
Thanks @jemc!
